src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise.service.ts

```diff
@@ -82,6 +82,22 @@ export class FileUploadExerciseService {
         return this.http.delete(`${this.resourceUrl}/${exerciseId}`, { observe: 'response' });
     }
 
+    /**
+     * Re-evaluates and updates an file upload exercise.
+     * @param fileUploadExercise that should be updated of type {FileUploadExercise}
+     * @param req optional request options
+     */
+    reevaluateAndUpdate(fileUploadExercise: FileUploadExercise, req?: any): Observable<EntityResponseType> {
+        const options = createRequestOption(req);
+        let copy = this.exerciseService.convertDateFromClient(fileUploadExercise);
+        copy = this.exerciseService.setBonusPointsConstrainedByIncludedInOverallScore(copy);
+        copy.categories = this.exerciseService.stringifyExerciseCategories(copy);
+        return this.http.put<FileUploadExercise>(`${this.resourceUrl}/re-evaluate`, copy, { params: options, observe: 'response' }).pipe(
+            map((res: EntityResponseType) => this.exerciseService.convertDateFromServer(res)),
+            map((res: EntityResponseType) => this.exerciseService.convertExerciseCategoriesFromServer(res)),
+        );
+    }
+
```

Comment:
Can you put this into a generic function at exercise service? It is the same for all exercise types.

Reply:
I thought a lot about it, I mean it. well, if we had general exercise service to update, i could also make generic re-evaluate service. I needed to call corresponding updateXExercise after re-evaluation has been done. So, I thought separate re-evaluate endpoints make sense in this case. If I put only this part into generic exercise service, I have to look at the exercise type and update the path for each exercise, then need to invoke related resource. Instead of it, this seems better to me. its open to discuss...